Guides/Transactions.md

```diff
@@ -0,0 +1,111 @@
+
+**Note**: Transactions only work with MongoDB replica sets (v4.0+) and sharded clusters (v4.2+).
+
+## Examples
+
+### Transaction that Atomically Moves a `Document` from One `MongoCollection` to Another
+```swift
+let elg = MultiThreadedEventLoopGroup(numberOfThreads: 4)
+
+defer {
+    // free driver resources
+    client.syncShutdown()
+    cleanupMongoSwift()
+
+    // shut down EventLoopGroup
+    try? elg.syncShutdownGracefully()
+}
+
+let client = try MongoClient(using: elg)
+let session = client.startSession()
+
+let db = client.db("test")
+let srcColl = db.collection("src")
+let destColl = db.collection("coll")
+let docToMove: Document = ["hello": "world"]
+
+session.startTransaction().flatMap { _ in
+    srcColl.deleteOne(docToMove, session: session)
+}.flatMap { _ in
+    destColl.insertOne(docToMove, session: session)
+}.flatMap { _ in
+    session.commitTransaction()
+}.whenFailure { error in
+    session.abortTransaction()
+    // handle error
+}
+```
+
+### Transaction with Custom Transaction Options
+```swift
+let elg = MultiThreadedEventLoopGroup(numberOfThreads: 4)
+
+defer {
+    // free driver resources
+    client.syncShutdown()
+    cleanupMongoSwift()
+
+    // shut down EventLoopGroup
+    try? elg.syncShutdownGracefully()
+}
+
+let client = try MongoClient(using: elg)
+let session = client.startSession()
+
+let txnOpts = TransactionOptions(
+    maxCommitTimeMS: 30,
+    readConcern: ReadConcern(.local),
+    readPreference: ReadPreference.primaryPreferred,
+    writeConcern: try WriteConcern(w: .majority)
+)
+
+session.startTransaction(options: txnOpts).flatMap { _ in
+    // do something
+}.flatMap { _ in
+    session.commitTransaction()
+}.whenFailure { error in
+    session.abortTransaction()
+    // handle error
+}
+```
+
+### Transaction with Default Transaction Options
+```swift
+let elg = MultiThreadedEventLoopGroup(numberOfThreads: 4)
+
+defer {
+    // free driver resources
+    client.syncShutdown()
+    cleanupMongoSwift()
+
+    // shut down EventLoopGroup
+    try? elg.syncShutdownGracefully()
+}
+
+let txnOpts = TransactionOptions(
+    maxCommitTimeMS: 30,
+    readConcern: ReadConcern(.local),
+    readPreference: ReadPreference.primaryPreferred,
+    writeConcern: try WriteConcern(w: .majority)
+)
+
+let client = try MongoClient(using: elg)
+let session = client.startSession(options: ClientSessionOptions(defaultTransactionOptions: txnOpts))
+
+session.startTransaction().flatMap { _ in
+    // do something
+}.flatMap { _ in
+    session.commitTransaction()
+}.whenFailure { error in
+    session.abortTransaction()
+    // handle error
+}
+```
+
+Note: Any transaction options provided directly to `startTransaction()` override the default transaction options for the session. More so, the default transaction options for the session override any options inherited from the client.
+
```

Comment:
The [error handling examples from the spec](https://github.com/mongodb/specifications/blob/master/source/transactions/transactions.rst#error-reporting-and-retrying-transactions) would be useful to have here. We can also link in the error handling guide's description of the various labels to this section once added.

Reply:
I'll make a note to link the error handling examples when [SWIFT-347](https://jira.mongodb.org/browse/SWIFT-347) is pushed. I've currently got a PR open for these changes.

Reply:
Filed [SWIFT-776](https://jira.mongodb.org/browse/SWIFT-776) to cover this work.